v3/src/data-interactive/handlers/di-handler-utils.ts

```diff
@@ -55,4 +61,13 @@ export function updateAttribute(attribute: IAttribute, value: DIAttribute, dataC
       metadata?.setIsHidden(attribute.id, value.hidden)
     }
   }
+  if (value?.colormap != null) {
+    if (dataContext) {
+      const metadata = getSharedCaseMetadataFromDataset(dataContext)
+      const categorySet = metadata?.getCategorySet(attribute.id)
+      Object.entries(value?.colormap as DICategoryColorMap || {}).forEach(([category, color]) => {
+        categorySet?.setColorForCategory(category, color || categorySet.colorMap[category] || "")
+      })
+    }
+  }
```

Comment:
You should reformat this block and the one above it (`value.hidden`) so they are both under a single `if (dataContext)` conditional, rather than duplicating a bunch of code.